config/default.js

```diff
@@ -0,0 +1,189 @@
+const {
+  VERSIONS,
+  RESOURCES
+} = require('@asymmetrik/node-fhir-server-core').constants;
+
+// // Set up whitelist
+// let whitelist_env = env.WHITELIST && env.WHITELIST.split(',').map(host => host.trim()) || false;
+//
+// // If no whitelist is present, disable cors
+// // If it's length is 1, set it to a string, so * works
+// // If there are multiple, keep them as an array
+// let whitelist = whitelist_env && whitelist_env.length === 1
+// 	? whitelist_env[0]
+// 	: whitelist_env;
+
+//
+// let supportedResources = [RESOURCES.ACCOUNT,
+//                           RESOURCES.PATIENT,
+//                           RESOURCES.OBSERVATION,
+//                           RESOURCES.MEDICATIONSTATEMENT,
+//                           RESOURCES.ADVERSEEVENT,
+//                           RESOURCES.ALLERGYINTOLERANCE,
+//                           RESOURCES.IMMUNIZATION,
+//                           RESOURCES.MEDICATIONADMINISTRATION,
+//                           RESOURCES.CONDITION,
+//                           RESOURCES.DEVICE,
+//                           RESOURCES.PROCEDURE,
+//                           RESOURCES.PRACTITIONER,
+//                           RESOURCES.ENCOUNTER,
+//                           RESOURCES.MEDICATIONREQUEST];
+// let profiles = {};
+// for (var i in supportedResources){
+//   let profile = supportedResources[i];
+//   profiles[profile] = { service: new PassThroughService(profile),
```

Comment:
I see the issue this previous approach caused in terms of circular dependencies, but would it also have been possible to just add config as a parameter to the PassThroughService constructor? I'm fine with either approach so no need to rework it again, just want to call out that this approach is easier when adding new functions to single controllers, the other approach would be easier when adding new controllers.

Reply:
It could have been architected in a way that did not require breaking out the controllers.   It was a trade off but I figure, for the reason that you already highlighted, that the ability to add specific functionality to individual controllers was a better approach.  I also wanted to start moving toward the config being more declarative without as much processing involved in it.